034-default-to-restricted-pod-security-profile.md

```diff
@@ -0,0 +1,157 @@
+If such tooling conflicts with the _restricted_ profile, such users would need to configure the security context in the custom resources.
+We have currently no way to know how many users use such tooling. 
+
+#### Installation files not working on OpenShift
+
+When we add the _restricted_ security context to the installation files of the operators, it would be possible to install them out of the box only on Kubernetes.
+OpenShift users would need to edit them first to remove / update the security context.
+This applies only to the YAML files.
+Operator Hub has separate sources for OpenShift and Kubernetes.
+And in the Helm Chart, there is an option to easily change the security context without editing the Helm Chart.
+
+### Benefits
+
+This proposal starts with focus on Pod Security Standards.
+However, tightening the security is in general useful even when the security profiles are not restricted.
+It would improve the security of our pods in all environments.
+
+### Feature Gate
+
+Due to the challenges described above, we will use Feature Gate to introduce this.
+With this feature gate disabled, the _restricted_ security context will not be set.
+With it enabled, the _restricted_ security context will be used by default.
+The security context will be added to the installation files only when the feature gate moves to beta and is enabled by default.
+The following table shows the expected graduation of the feature gate:
+
+| Phase | Strimzi versions       | Default state                                                       |
+|:------|:-----------------------|:--------------------------------------------------------------------|
+| Alpha | 0.28, 0.29             | Disabled by default, security context not set in installation files |
+| Beta  | 0.30, 0.31             | Enabled by default  security context not set in installation files  |
+| GA    | 0.32 and newer         | Enabled by default (without possibility to disable it)              |
+
+Just to note:
+* Even after this feature gate graduates to GA, users will be still able to set the security context using the custom resources.
+  So it will not be permanent, it will be just harder to override the defaults
+* On OpenShift, the new default values will not be set even after the graduation of the feature gate.
+
+## Rejected alternatives
+
+### Permanent switch instead of Feature Gate
+
+Instead of using a Feature Gate to introduce this feature, we can use a regular configuration option to enable or disable this feature.
+This would work similarly to how the flag for disabling the Network Polices was implemented.
+We would only need to decide what should be the default value (enabled or disabled by default).
+
+The disadvantage of this would be that we would have to maintain this permanently in the code and not just while the feature gate is used.
+(However we might need to do that for OpenShift anyway, this would be just second flag to check)
```

Comment:
It would also allow for a bit more configurability in the kube-with-custom-security-tooling case. E.g. could configure a container SC that was provided at the operator level, avoiding the need to add it to every CR. But the possible need to have one per container type doesn't make it _very_ attractive as an alternative.

Reply:
Actually, what I had in mind here was more environment variable like `STRIMZI_SECURITY_PROFILE=restricted`. Not passing the whole security context. But Passing the full security context YAML as the env var might be another option. Not sure how practical since the context might need to differ slightly for pods using persistent storage (Zoo, Kafka) and stateless pods (Connect, EO, Bridge, ...).

Reply:
If the flag option is suitable for OpenShift, maybe it's easier to take this approach from the outset

Reply:
The flag would not work as _more secure by default_. But it would make the installation easier since it would not need two sets of installation files etc.